.github/workflows/haskell.yml

```diff
@@ -17,6 +20,9 @@ jobs:
   build:
     runs-on: ${{ matrix.os }}
 
+    # Always run on master
+    if: github.event_name != 'create' || github.ref_name == 'master'
```

Comment:
What's the `github.event_name != 'create'` for?

Are we saying run only if this is not for a `create` event (which as I understand happens when a branch or tag is created)?

What are we admitting and excluding in the non-`master` case?

Reply:
I agree, this conditional here seems somewhat odd.